File: src/main/java/com/educlaas/sociallogin/service/StoreService.java
package com.educlaas.sociallogin.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.educlaas.sociallogin.dao.Stores;
import com.educlaas.sociallogin.repository.StoreRepository;


@Service
public class StoreService {

	@Autowired
	private StoreRepository storeRepository;

	public String addStore(Stores stores) {
		storeRepository.save(stores);
		System.out.println(stores);
		return "Store Added successfully";
	}
	
	public List<Stores> getAllStores() {
		List<Stores> stores = storeRepository.findAll();
		return stores;
	}
	
	public List<Stores> searchStores(String keyword){
		List<Stores> list = storeRepository.searchStore(keyword);
		return list;
	}
}

File: src/service/StoreService.js
import axios from 'axios';

const API_BASE_URL = "http://localhost:8080/online"

class StoreService{
    
    addStores(stores){
        return axios.post(API_BASE_URL+"/addstore", stores)
    }

    getAllStores(){
        return axios.get(API_BASE_URL+"/viewallstores")
    }

    searchStore(keyword){
        return axios.get(API_BASE_URL+"/search/"+keyword)
    }
    
}

export default new StoreService();